profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfileImageLogicImpl.java

```diff
@@ -1129,6 +1133,22 @@ private String getAvatarInitialsColor(String displayName) {
 		return profileAvatarColors[colorIndex];
 	}
 
+	public boolean isPicEditorEnabled() {
+		String userUuid = sakaiProxy.getCurrentUserId();
+		UserProfile userProfile = (UserProfile) profileLogic.getUserProfile(userUuid);
+		ProfilePreferences prefs = preferencesLogic.getPreferencesRecordForUser(userUuid);
+		// -is picture changing disabled? (property, or locked).
+		// -if using official images, is the user allowed to select an alternate?
+		//  or have they specified the official image in their preferences.
+		// -last condition prevents the editor being enabled when only official img is allowed by property. 
+		if((!sakaiProxy.isProfilePictureChangeEnabled() || userProfile.isLocked()) && !sakaiProxy.isSuperUser() ||
+				sakaiProxy.isOfficialImageEnabledGlobally() && (!sakaiProxy.isUsingOfficialImageButAlternateSelectionEnabled() || prefs.isUseOfficialImage()) ||
+					sakaiProxy.isProfilePictureChangeEnabled() && sakaiProxy.isUsingOfficialImage()) {
+			return false;
```

Comment:
Moved logic to its own method as now will be used in various places.

Added an extra condition to avoid the following situation:
When `profile2.picture.change.enabled=true` & `profile2.picture.type=official`
Student apparently could change its profile pic, but after clicking _save_, the new img wouldn't apply.

Reply:
I find this complex, hard to read and hard to maintain, could we split this in simple conditions?